tools/gen-random-bout/gen-random-bout.cpp

```diff
@@ -58,80 +69,186 @@ static void push_range(KTest *b, const char *name, unsigned value) {
     srandom(seed);
-  else srandom(time(NULL) * getpid());
+  else
+    srandom(time(NULL) * getpid());
+
+  argv_copy = (char **)malloc((argc - 1) * sizeof(char *));
+  argv_copy[0] = argv[0];
+  for (i = 2; i < (unsigned)argc; ++i) {
+    argv_copy[i - 1] = argv[i];
+  }
 
   KTest b;
-  b.numArgs = argc;
-  b.args = argv;
+  b.numArgs = argc - 1;
+  b.args = argv_copy;
   b.symArgvs = 0;
   b.symArgvLen = 0;
 
   b.numObjects = 0;
   b.objects = (KTestObject *)malloc(MAX * sizeof *b.objects);
 
-  for(i = 2; i < (unsigned)argc; i++) {
-    if(strcmp(argv[i], "--sym-args") == 0) {
+  for (i = 2; i < (unsigned)argc; i++) {
+    if (strcmp(argv[i], "--sym-arg") == 0 || strcmp(argv[i], "-sym-arg") == 0) {
+      unsigned nbytes = getint(argv[++i]);
+
+      // A little different than how klee does it but more natural for random.
+      char arg[1024];
+      unsigned x = random() % (nbytes + 1);
+
+      sprintf(arg, "arg%d", total_args++);
+      push_random_obj(&b, arg, x, nbytes + 1);
+    } else if (strcmp(argv[i], "--sym-args") == 0 ||
+               strcmp(argv[i], "-sym-args") == 0) {
       unsigned lb = getint(argv[++i]);
       unsigned ub = getint(argv[++i]);
       unsigned nbytes = getint(argv[++i]);
 
-      narg = random() % (ub - lb) + lb;
-      push_range(&b, "range", narg);
+      if (ub < lb) {
+        fprintf(stderr, "--sym-args first argument should be no more than its "
+                        "second argument");
+        exit(1);
+      }
+
+      narg = random() % (ub - lb + 1) + lb;
+      push_range(&b, "n_args", narg);
 
-      while(narg-- > 0) {
+      while (narg-- > 0) {
         unsigned x = random() % (nbytes + 1);
 
         // A little different than how klee does it but more natural
         // for random.
-        static int total_args;
         char arg[1024];
 
         sprintf(arg, "arg%d", total_args++);
-        push_obj(&b, arg, x, nbytes+1);
-      }
-    } else if(strcmp(argv[i], "--sym-stdout") == 0) {
-      if(!sym_stdout) {
-        sym_stdout = 1;
-        push_obj(&b, "stdout", 1024, 1024);
-        push_obj(&b, "stdout-stat", sizeof(struct stat64), 
-                 sizeof(struct stat64));
+        push_random_obj(&b, arg, x, nbytes + 1);
       }
-    } else if(strcmp(argv[i], "--sym-files") == 0) {
+    } else if (strcmp(argv[i], "--sym-stdout") == 0 ||
+               strcmp(argv[i], "-sym-stdout") == 0) {
+      sym_stdout = 1;
+    } else if (strcmp(argv[i], "--sym-stdin") == 0 ||
+               strcmp(argv[i], "-sym-stdin") == 0) {
+      stdin_size = getint(argv[++i]);
+    } else if (strcmp(argv[i], "--sym-files") == 0 ||
+               strcmp(argv[i], "-sym-files") == 0) {
       unsigned nfiles = getint(argv[++i]);
       unsigned nbytes = getint(argv[++i]);
 
-      while(nfiles-- > 0) {
-        push_obj(&b, "file", nbytes, nbytes);
-        push_obj(&b, "file-stat", sizeof(struct stat64), sizeof(struct stat64));
+      total_files = 0;
+      while (nfiles-- > 0) {
```

Comment:
No max check for file_sizes. Also 1024 is quite large...

Reply:
@251 Added checks at access points of `file_sizes`.

Reply:
@251 Reduced 1024 to 256. I suppose we don't want the value to be too small here as some programs can accept a large number of input files.